cmd/root.go

```diff
@@ -41,7 +33,49 @@ var rootCmd = &cobra.Command{
 func Execute() {
 	cobra.OnInitialize(initHover)
 	if err := rootCmd.Execute(); err != nil {
-		log.Errorf("Command failed: %v", err)
+		logx.Errorf("Command failed: %v", err)
 		os.Exit(1)
 	}
 }
+
+func initHover() {
+	var (
+		colorsOpt = logx.OptionNoop
+	)
+
+	if colors {
+		colorsOpt = logx.OptionColorize
+	}
+
+	switch verbose {
+	case 2:
+		tracex.Tune(
+			colorsOpt,
+			logx.OptionStderr,
+		)
+		tracex.Println("trace logging enabled")
+		fallthrough
+	case 1:
+		debugx.Tune(
+			colorsOpt,
+			logx.OptionStderr,
+		)
+		debugx.Println("debug logging enabled")
+		fallthrough
+	default:
+		// informational logging only.
+		logx.Tune(colorsOpt)
```

Comment:
Why have debug and trace in separate packages? I think it would be cleaner if they're added to the `log` package as `.Debug(..)` and `.Trace(..)`. Then we just have to tell the log package whether it should print out calls to those functions or just no-op, depending on the --verbose level.

Reply:
its cleaner. examples:

```golang
tracex.Output(1,"foo")
tracex.Println("foo")
tracex.Printf("%s", "foo")
debugx.Output(1, "foo")
debugx.Println("foo")
debugx.Printf("%s", "foo")
logx.Output(1, "foo")
logx.Println("foo")
logx.Printf("%s", "foo")
```

```golang
log.Debug().Output(1, "foo")
log.Debug().Println("foo")
log.Debug().Printf("%s", "foo")
log.Trace().Output(1, "foo")
log.Trace().Println("foo")
log.Trace().Printf("%s", "foo")
```

Reply:
I think we should just add this to log.go

```go
// Debugf prints prints debug information when verbose is enabled.
func Debugf(part string, parts ...interface{}) {
	if !verbose {
		return
	}
	hoverPrint()
	fmt.Println(Au().Colorize(fmt.Sprintf(fmt.Sprintf("%v", part), parts...), aurora.GreenFg).String())
}
```

Maybe with a different color, but that's it.
Changing log.go to print to stderr is good though, it should be printing log output to stderr indeed.
I would prefer to keep that simple and use:
```go
fmt.Fprintf(os.Stderr,  ...
```

Reply:
Usage examples:
```go
log.Error("Failed to do blah: %v", err)
log.Infof("It worked!")
log.Debugf("Going to start command %s", cmd.String())
```

Reply:
we lose the file information and continue to log to stdout which is less than ideal (for a number of reasons) and don't really support varying the noise level of the information.